variables.tf

```diff
@@ -1,9 +1,67 @@
+##############################################################################
+# Common variables
+##############################################################################
+
+variable "resource_group_id" {
+  type        = string
+  description = "The resource group ID where resources will be provisioned."
+}
+
+variable "region" {
+  description = "The region to provision the resources."
+  type        = string
+  default     = "us-south"
+
+  validation {
+    condition     = contains(["us-east", "us-south", "eu-gb", "au-syd"], var.region)
+    error_message = "Value for region must be one of the following: ${join(", ", ["us-east", "us-south", "eu-gb", "au-syd"])}"
+  }
+}
+
 ########################################################################################################################
-# Input Variables
+# App Config Instance Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "app_config_name" {
+  type        = string
+  description = "Name for the App Configuration service instance"
+}
+
+variable "app_config_plan" {
+  type        = string
+  description = "Plan for the App Configuration service instance"
+  default     = "standardv2"
+
+  validation {
+    condition     = contains(["lite", "standardv2", "enterprise"], var.app_config_plan)
+    error_message = "Value for plan must be one of the following: \"lite\", \"standard\", or \"enterprise\"."
```

Comment:
Is it `standard` or `standardv2`? The condition says `standardv2` bit the error message has `standard`

Reply:
"standardv2" is correct, updated error message to reflect. I also had considered allowing "standard" and then appending the "v2" if that plan was chosen but decided against it as it seems better to require the inputs match what the provider expects.